scripts/list_boards.py

```diff
@@ -251,14 +310,21 @@ def find_v2_boards(args):
     root_args = argparse.Namespace(**{'soc_roots': args.soc_roots})
     systems = list_hardware.find_v2_systems(root_args)
 
-    boards = []
+    boards = {}
+    board_extensions = []
     board_files = []
-    for root in unique_paths(args.board_roots):
-        board_files.extend((root / 'boards').rglob(BOARD_YML))
+    if args.board_dir:
+        board_files = [d / BOARD_YML for d in args.board_dir]
+    else:
+        for root in unique_paths(args.board_roots):
+            board_files.extend((root / 'boards').rglob(BOARD_YML))
 
     for board_yml in board_files:
-        b = load_v2_boards(args.board, board_yml, systems)
-        boards.extend(b)
+        b, e = load_v2_boards(args.board, board_yml, systems)
+        boards.update(b)
```

Comment:
Should we make sure that no two boards can share the same `name`, both here and in `load_v2_boards()`?
Especially considering that this is no longer checked in CMake.

Reply:
good point.
A board can be extended but two boards should not both create conflicting boards.

Reply:
Fixed, if duplicate boards are now discovered then list_boards will print an error:
https://github.com/zephyrproject-rtos/zephyr/blob/41561f164a8f7bc1e697fd1f7942700dd4b54854/scripts/list_boards.py#L327-L330